### PR TITLE
Stress test update

### PR DESCRIPTION
**Pull request by ZiyanShi**
### What problem does this PR solve?

Refactoring stress test and add stress test for string data type.

Tests 

- Unit test
- Stress test

Side effects

### CODE REVIEW

File: tests/stress_test.cpp [hunk 1]
```diff
@@ -688,106 +696,120 @@ class EngineStressTest : public EngineTestBase {
...
   }
 }
 
 class EngineHotspotTest : public EngineTestBase {
-private:
+protected:
   virtual void SetUpParameters() override final {
     /// Default configure parameters
     do_populate_when_initialize = false;
     // 16GB PMem
```
Comment (sherry-1001) [inline tests/stress_test.cpp:770]: 64GB PMem

File: tests/stress_test.cpp [hunk 2]
```diff
@@ -17,332 +18,393 @@
...
+
+class StringOperator {
+  kvdk::Engine *engine;
+  CollectionNameType collection_name;
+
+public:
+  StringOperator() = delete;
+  // For convenince, introducing empty collection_name for global anonymous
+  // collection
+  StringOperator(kvdk::Engine *e, CollectionNameType cn)
```
[the remark this answers is not in the document]
Comment (ZiyanShi) [reply tests/stress_test.cpp]: No need, and should not.
Different collections can share same name. Should distinguish between the them.

File: tests/stress_test.cpp [hunk 3]
```diff
@@ -17,332 +18,393 @@
...
+};
 
-  std::unordered_multimap<StringView, StringView> possible_kv_pairs_copy{
-      possible_kv_pairs};
+enum class IteratingDirection { Forward, Backward };
 
-  auto u_iter = engine->NewUnorderedIterator(collection_name);
+template <typename EngineOperator> class DataBaseStateTracker {
+public:
+  struct VState {
```
[the remark this answers is not in the document]
Comment (ZiyanShi) [reply tests/stress_test.cpp]: Changed to StateAndValue. That's more clear.

File: tests/stress_test.cpp [hunk 4]
```diff
@@ -17,332 +18,393 @@
...
-      CheckKVPair(key, value_got, possible_kv_pairs);
-
-      possible_kv_pairs.erase(key);
-      n_removed_possible_kv_pairs =
-          n_total_possible_kv_pairs - possible_kv_pairs.size();
-      if ((n_removed_possible_kv_pairs > old_progress + 1000) ||
-          possible_kv_pairs.empty()) {
-        progress_iterating.Update(n_removed_possible_kv_pairs);
-        old_progress = n_removed_possible_kv_pairs;
+    for (size_t tid = 0; tid < n_thread; tid++) {
```
[the remark this answers is not in the document]
Comment (ZiyanShi) [reply tests/stress_test.cpp]: renamed

File: tests/stress_test.cpp (hunk 1 above)
Comment (ZiyanShi) [reply tests/stress_test.cpp:784]: done

**Review by sherry-1001 — APPROVED** [review]

**Review by JiayuZzz — APPROVED** [review]

**Pull request merged**